### PR TITLE
[tape] add `tape/lib/test` types

### PR DESCRIPTION
**Pull request by ljharb**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [ ] Test the change in your own code. (Compile and run.)
- [ ] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [ ] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [ ] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [ ] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If changing an existing definition:
- [ ] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>

### CODE REVIEW

**Comment by ljharb** [comment]
cc @sandersn any tips here?

File: types/tape/OTHERFILES.txt
Comment (sandersn) [inline types/tape/OTHERFILES.txt:1]: this should be `lib/tape.d.ts` to match the filename (or maybe the filename should change to be test.d.ts)

File: types/tape/tape-tests.ts
Comment (sandersn) [inline types/tape/tape-tests.ts:8]: this doesn't do anything in a .ts file and should be removed

**Review by sandersn — CHANGES_REQUESTED** [review]

**Comment by sandersn** [comment]
Sorry I didn't notice this before, but the filename is OTHER_FILES.txt not OTHERFILES.txt

**Comment by sandersn** [comment]
But also, it looks like the test usage counts and OTHER_FILES isn't needed.

File: types/tape/tape-tests.ts
Comment (sandersn) [inline types/tape/tape-tests.ts:3]: the linter is crashing because it didn't expect the double semicolon. I'll file a bug on `@definitelytyped/header-parser`

Comment (ljharb) [reply types/tape/tape-tests.ts:3]: oh whoops, thanks.

Comment (ljharb) [reply types/tape/tape-tests.ts]: Note that it also tripped up on the unnecessary semicolon i had after the `}` in `declare module 'blah' {};`

**Comment by sandersn** [comment]
I skipped filing a bug on the linter since it was a simple omission from a recent change: https://github.com/microsoft/DefinitelyTyped-tools/pull/635

**Comment by DangerBotOSS** [comment]
Inspecting the JavaScript source for this package found some properties that are not in the .d.ts files.
The check for missing properties isn't always right, so take this list as advice, not a requirement.
## tape ([<kbd>unpkg</kbd>](https://unpkg.com/browse/tape@latest/))
was missing the following properties:
1. getHarness
1. Test
1. test
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 3ca161c066811d8854d83ac9b67d7cb377245819
</p>

**Comment by ljharb** [comment]
Added some extra commits per Dangerbot ¯\\\_(ツ)_/¯

**Review by sandersn — APPROVED** [review]
Looks good structurally, but I like to get a review from one of the other maintainers for popular packages.

**Comment by ljharb** [comment]
ping @Bartvds, @sodatea, @DennisSchwartz, @mikehenrty, @rostrowski - can any of you take a look?

**Review by mikehenrty — APPROVED** [review]
Changes look fine overall. Added a small suggestion, but I'll approve for now anyway.

**Comment by ljharb** [comment]
Ready to merge

**Pull request merged**
